sw/airborne/math/pprz_stat.c

```diff
@@ -73,18 +78,22 @@ int32_t variance_i(int32_t *array, uint32_t n_elements)
  */
 int32_t covariance_i(int32_t *array1, int32_t *array2, uint32_t n_elements)
 {
-  // Determine means for each vector:
-  float sumX = 0.f, sumY = 0.f, sumXY = 0.f;
-
-  // Determine the covariance:
-  uint32_t i;
-  for (i = 0; i < n_elements; i++) {
-    sumX += (float)array1[i];
-    sumY += (float)array2[i];
-    sumXY += (float)(array1[i]) * (float)(array2[i]);
+  if (n_elements == 0) {
+    // Note that something else is wrong if you want the covariance of 0 samples.
+    return 0;
+  } else {
```

Comment:
Writing it like this is not good practice as the second return only happens if the else is called but the return uses things that are defined inside the else, not good.

You should have the if with the n_elements then the rest follow without the else and brackets.